pkg/steps/template.go

```diff
@@ -48,6 +57,61 @@ type templateExecutionStep struct {
 	subTests []*junit.TestCase
 }
 
+func injectResourcesToTemplatePods(template *templateapi.Template, resources api.ResourceConfiguration) error {
+	if resources != nil {
```

Comment:
nit: passing an argument into a function and then using it to determine if work should be done is a smell -- don't call the function at all if you don't have resources to set